src/main/java/com/bloxbean/cardano/client/crypto/bip32/HdKeyGenerator.java

```diff
@@ -206,6 +212,58 @@ public HdKeyPair getChildKeyPair(HdKeyPair parent, long child, boolean isHardene
         return key;
     }
 
+    /**
+     * Derive the public child key from HD parent public key
+     *
+     * @param parent     the parent key
+     * @param child      the child index
+     * @return
+     */
+    public HdPublicKey getChildPublicKey(HdPublicKey parent, long child) {
+        HdPublicKey publicKey = new HdPublicKey();
+        byte[] AP = parent.getKeyData();
+
+        byte[] pChain = parent.getChainCode();
+        byte[] childNumber = BytesUtil.ser32(child);
+
+        //prefix 0x02 for child public key
+        byte[] ApLE = serializeUnsignedLE256(parseUnsignedLE(AP));
+        byte[] data = BytesUtil.merge(new byte[]{2}, ApLE, BytesUtil.ser32LE(child));
+        byte[] Z = Hmac.hmac512(data, pChain);
+
+        //prefix 0x03 for child chain code
+        data[0] = 3;
+        byte[] c = Hmac.hmac512(data, parent.getChainCode());
+
+        //truncate to right 32 bytes for child chain code
+        c = Arrays.copyOfRange(c, 32, 64);
+
+        // split into left (28 bytes) /right (for child public key)
+        byte[] ZL = Arrays.copyOfRange(Z, 0, 28);
+//      byte[] ZR = Arrays.copyOfRange(Z, 32, 64);
+
+        //Ai ← AP + [8ZL]B,
+        BigInteger kLiBI = parseUnsignedLE(ZL)
+                .multiply(BigInteger.valueOf(8));
+
+        byte[] kLi = serializeUnsignedLE256(kLiBI);
+
+        GroupElement gp1 = new GroupElement(ED25519SPEC.getCurve(), AP);
+        gp1 = gp1.toCached();
+        GroupElement groupElement = ED25519SPEC.getB().scalarMultiply(kLi).add(gp1);
+        //TODO -- If Ai is the identity point (0, 1), discard the child
```

Comment:
If possible, please do this check. Otherwise, you can leave that for now.

Reply:
I will need to do some research into the identity point.

Reply:
Thanks. I will merge this PR for now. You can impl identity point check through another PR.